# mk-hicup-mapper

Run the HiCUP mapper on truncated FASTQ files to obtain aligned SAM files


## About the alignment ##

From the [HiCUP documents](https://www.bioinformatics.babraham.ac.uk/projects/hicup/scripts_description/#Mapper), these are the Bowtie2 Mapping Parameters:

* very-sensitive: slower but a more sensitive and more accurate option.
* no-unal: suppress SAM records for reads that failed to align.
* threads: number of threads specified by the user divided by the number of files processed.
* reorder: ensure the read output order is the same as the input order when multi-threading.

Bowtie2 does not have a direct equivalent of the -m 1 option available in the original Bowtie. Therefore, to identify and filter out multi-mapping reads, HiCUP processes the SAM file generated by Bowtie2. 

A reads is considered as uniquely mapping if the quality score is greater than or equal to 30 and either i) the read cannot be mapped to another location or ii) if the read can be mapped to other locations, then the difference in quality score between this hit and the next-best match should be at least 10 (as reported in the Bowtie2 SAM tags "AS" and "XS"). Before HiCUP version 0.6.1, a unique-mapping read was defined simply as having no next-best hit when using Bowtie2 as the aligner.

